light_curves/code/sample_selection.py

```diff
@@ -372,68 +370,47 @@ def get_paper_sample(paper_link,label,coords,labels,verbose=1):
     if verbose:
         print("number of sources added from "+str(label)+" :"+str(len(paper_coords)))
 
-    
-#and now a function to remove duplicates from the coordinate list and attach an object id
-def clean_sample(skycoordslist, labels, verbose=1):
+
+def clean_sample(coords_list, labels_list, verbose=1):
     """Makes a unique sample of skycoords and labels with no repeats. Attaches an object ID to the coords.
-    
+
     Parameters
     ----------
-    skycoordslist : list
+    coords_list : list
         list of Astropy SkyCoords derived from literature sources
-    lables : list
+    labels_list : list
         List of the first author name and publication year for tracking the sources
     verbose : int, optional
         Print out the length of the sample after applying this function
-        
+
     Returns
     -------
-    coords_list : list of tuples
-        coords input cleaned of duplicates, with an object ID attached. Tuples contain (objectid, skycoords).
-    labels_list : list
-        labels associated with coords_list
+    sample_table : `~astropy.table.Table`
+        sample cleaned of duplicates, with an object ID attached.
     """
-    #first turn the skycoord list into a table to be able to access table functions in astropy
-    t = Table([skycoordslist, labels, np.arange(0, len(skycoordslist), 1)], names=['sc', 'label', 'idx'])
 
-    #now join the table with itself within a defined radius
-    tjoin = join(t, t, keys='sc', join_funcs={'sc': join_skycoord(0.005 * u.deg)})
-    #this join will return 4 entries for each redundant coordinate = 
-    #1 for the match with itself and 1 for the match with the similar enough coord target then the same thing again for the match
-    #but all of these 4 will have the same id in the new 'sc_id' column made by the join function
+    sample_table = Table([coords_list, labels_list], names=['coord', 'label'])
+
+    # now join the table with itself within a defined radius.
+    # We keep one set of original column names to avoid later need for renaming
+    tjoin = join(sample_table, sample_table, keys='coord',
+                 join_funcs={'coord': join_skycoord(0.005 * u.deg)},
+                 uniq_col_name='{col_name}{table_name}', table_names=['', '_2'])
 
-    #keep only those entries in the resulting table which are unique 
-    uniquerows = table.unique(tjoin, keys = 'sc_id')
+    # this join will return 4 entries for each redundant coordinate:
+    # 1 for the match with itself and 1 for the match with the similar
+    # enough coord target then the same thing again for the match, but all of
+    # these 4 will have the same id in the new 'coords_id' column
+
+    # keep only those entries in the resulting table which are unique
+    uniqued_table = unique(tjoin, keys='coord_id')['coord_id', 'coord', 'label']
+    uniqued_table.rename_column('coord_id', 'object_id')
```

Comment:
I recommend that we use `objectid` instead of `object_id` because the former is how it's named in the light curve dataframes and using the same name will help avoid confusion.

Reply:
OK, I can change that back. Going forward though I think it would be nice to follow some more widely accepted naming scheme, snake case for variables, and full words, etc.

Reply:
Generally speaking, I'm all in favor of snake case, clear and descriptive variable names, etc. It would be nice to be able to clean up and standardize the code before it's so deeply developed that changes become a real headache. It's obviously important to let the scientists blaze ahead and hack away in a way that works for them so they can flesh out their ideas. But maybe as a group we could develop a workflow where code gets reviewed and revised earlier on, after they've been able to develop the idea a little bit but before there's so much code that changing something is a real pain.

Reply:
(Perhaps more importantly, this might help us catch and correct structural issues early on, like the split between `coords_list` and `label_list`.)

Reply:
Or the manual looping issues. Though one wonders how much drag early reviews would add to the process, after all a lot of the original code got condensed down already and at this point, and working with the `code_src` files is way nicer and adds fewer code conflicts than it would have been with the first drafts of the notebook when everything was way too fluid.

Reply:
> Though one wonders how much drag early reviews would add to the process

Ya, I agree. It would be easy for this to just end up making the work harder and people more frustrated. Might help to get clear as a group about what should be reviewed at this stage and for the reviewers not to go beyond that scope. Maybe an initial list is:
- structural issues like `coords_list` and `labels_list`
- error-prone patterns like manual looping
- names of things that will be used extensively like `objectid`

Reply:
> error-prone patterns like manual looping

I tried to pull these out to e.g. #157. Will rebase this PR to be based on top of that, so review can look at only the diff commits.

But some of the changes are difficult to recouple as they are connected to e.g. the structural changes, or only make sense to do once those are done. Sadly even some of the pep8 stuff is in this category, e.g. why fix whitespace when the given codeblock is known to go away in a subsequent commit.

Reply:
This is all good for me to see.  I have added this as an agenda item for wednesday's team meeting.